ertviz/controllers/response_correlation_controller.py

```diff
@@ -0,0 +1,273 @@
+        _layout.update(dict(showlegend=False))
+        fig.update_layout(_layout)
+        fig.add_shape(
+            type="line",
+            x0=x_axis[corr_xindex],
+            y0=0,
+            x1=x_axis[corr_xindex],
+            y1=1,
+            yref="paper",
+            line=dict(color="rgb(30, 30, 30)", dash="dash", width=3),
+        )
+        fig.update_layout(clickmode="event+select")
+        return fig
+
+    @app.callback(
+        Output(
+            {"id": parent.uuid("response-scatterplot"), "type": parent.uuid("graph")},
+            "figure",
+        ),
+        [
+            Input(parent.uuid("correlation-store-xindex"), "modified_timestamp"),
+            Input(parent.uuid("correlation-store-parameter"), "modified_timestamp"),
+            Input(parent.uuid("parameter-selection-store-resp"), "modified_timestamp"),
+            Input(parent.uuid("parameter-selection-store-param"), "modified_timestamp"),
+            Input(parent.uuid("ensemble-selection-store"), "modified_timestamp"),
+        ],
+        [
+            State(parent.uuid("parameter-selection-store-param"), "data"),
+            State(parent.uuid("parameter-selection-store-resp"), "data"),
+            State(parent.uuid("ensemble-selection-store"), "data"),
+            State(parent.uuid("correlation-store-xindex"), "data"),
+            State(parent.uuid("correlation-store-parameter"), "data"),
+        ],
+    )
+    def update_response_parameter_scatterplot(
+        _,
+        __,
+        ___,
+        ____,
+        _____,
+        parameters,
+        responses,
+        ensembles,
+        corr_xindex,
+        corr_parameter,
+    ):
+
+        if not parameters or not ensembles or not responses:
+            raise PreventUpdate
+
+        parameter = corr_parameter if corr_parameter else parameters[0]
+        _plots = []
+        _resp_plots = {}
+        _param_plots = {}
+        _colors = {}
+        for ensemble_id, data in ensembles.items():
+            ensemble = load_ensemble(parent, ensemble_id)
+            # take the first selected response
+            y_data = ensemble.parameters[parameter].data_df()
+            response = ensemble.responses[responses[0]]
+            x_data = response.data_df().loc[corr_xindex]
+            style = deepcopy(assets.ERTSTYLE["response-plot"]["response-index"])
+            style["marker"]["color"] = data["color"]
+            _colors[str(ensemble)] = data["color"]
+            _plots += [
+                PlotModel(
+                    x_axis=x_data.values.flatten(),
+                    y_axis=y_data.values.flatten(),
+                    text="Mean",
+                    name=f"{repr(ensemble)}: {responses[0]}x{parameter}@{response.axis[corr_xindex]}",
+                    **style,
+                )
+            ]
+            _resp_plots[str(ensemble)] = x_data.values.flatten()
+            _param_plots[str(ensemble)] = y_data.values.flatten()
```

Comment:
Second time extracting values and flattening of those in 8 lines - should those be a variable?

Reply:
I need more clarification on this.